Qubit unitary decomposition (#7277)
PennyLaneAI/pennylane: doc/releases/changelog-dev.md

<h3>Improvements 🛠</h3>

> * Added decomposition to ``qml.QubitUnitary`` for any number of qubits.

Reviewer: ```suggestion
* A decomposition rule has been added to ``qml.QubitUnitary`` that works for any number of qubits.
```

Can we show what the decomposition rule is, maybe? You can print the decomposition rule with `print(qml.list_decomps(qml.QubitUnitary)[index])` and it will show the quantum function + resources registered :) 

Is there a benefit to using this decomposition outside of it being general to the number of qubits?

Reviewer 2: It's cheaper than using the default select decomposition in the case where we have multiple Multiplexers in the circuit because we can prepare a single resource state and re-use it for each multiplexer.

Reviewer: ```suggestion
* A more resource efficient decomposition rule has been added to :class:`~.QubitUnitary` that works for 
  any number of qubits.
```

<h3>Improvements 🛠</h3>

> * A more resource efficient decomposition rule has been added to :class:`~.QubitUnitary` that works for 
>   any number of qubits.

Reviewer 3: ```suggestion
* :class:`~.QubitUnitary` now uses a more resource-efficient decomposition routine that works for 
  any number of qubits.
```

<h3>Improvements 🛠</h3>

* A more resource efficient decomposition rule has been added to :class:`~.QubitUnitary` that works for 
  any number of qubits.

>   [(#7277)](https://github.com/PennyLaneAI/pennylane/pull/7277)

Reviewer: Can we show what the decomposition rule is, maybe? You can print the decomposition rule with print(qml.list_decomps(qml.QubitUnitary)[index]) and it will show the quantum function + resources registered :)

Reviewer: Oh so this only works with the new system disabled?

Reviewer 2: Ya Issac, here Guillermo improved the compute_decomposition method instead of adding a decomposition rule and registering resources. I think we should change the wording here so that we don't say "decomposition rule" as this could cause confusion with other developers.
